Cargo.toml

```diff
@@ -43,6 +43,9 @@ time = "0.3.7"
 tokio = { version = "1.17", features = ["full"] }
 tower = "0.4"
 
+minitrace = "0.4.0"
```

Comment:
Simply adding `minitrace` here does nothing, we need to add instruments into our code.